features/keychain/module/keychain.js

```diff
@@ -44,30 +44,50 @@ export class Keychain {
     this.secp256k1 = secp256k1.create({ getPrivateHDKey: this.#getPrivateHDKey })
   }
 
-  #assertPrivateKeysUnlocked() {
-    assert(!this.#privateKeysAreLocked, 'private keys are locked')
+  #assertPrivateKeysUnlocked(seedIds) {
+    const locked = this.#checkPrivateKeysLocked(seedIds)
+    assert(!locked, 'private keys are locked')
   }
 
-  arePrivateKeysLocked() {
-    return this.#privateKeysAreLocked
+  #checkPrivateKeysLocked(seedIds) {
+    if (!seedIds?.size) {
+      return Object.values(this.#seedLockStatus).some(Boolean)
+    }
+
+    const existingSeeds = Object.entries(this.#seedLockStatus)
+    const checkingSeeds = existingSeeds.filter(([seedId]) => seedIds.has(seedId))
+
+    assert(checkingSeeds.length === seedIds.size, 'must pass in existing seeds')
+    return checkingSeeds.some(([, locked]) => locked)
```

Comment:
nit: maybe a tad easier to follow (maybe pass seedIds as array or use `[...seedIds]` if we keep the set)
```suggestion
    return seedIds.some((seedId) => { 
      assert(Object.hasOwn(this.#seedLockStatus, seedId), `cannot check lock state for unknown seed "${seedId}"`)
      return this.#seedLockStatus[seedId]
    })
```